Virtualbox 7.x will only build a lib named VBoxPython3.so, regardless of the
actual python version used when compiling.  Remove VBoxPython3m.so, we don't
use it.

Upstream added support for pyton up to 3.12.

--- a/src/libs/xpcom18a4/python/Makefile.kmk
+++ b/src/libs/xpcom18a4/python/Makefile.kmk
@@ -704,13 +704,6 @@
    VBoxPython3_DEFS      = $(filter-out VBOX_PYXPCOM_VERSIONED,$(VBoxPythonBase_DEFS)) Py_LIMITED_API=0x03030000
    VBoxPython3_INCS      = $(VBoxPythonBase_INCS) $(VBOX_PYTHON$(VBOX_PYTHON_LIMITED_API_VER)_INC)
    VBoxPython3_LDFLAGS.darwin = -undefined dynamic_lookup
-
-   ifneq ($(KBUILD_TARGET),darwin)
-    DLLS += VBoxPython3m
-    VBoxPython3m_EXTENDS  = VBoxPythonBase_m
-    VBoxPython3m_DEFS     = $(filter-out VBOX_PYXPCOM_VERSIONED,$(VBoxPythonBase_m_DEFS)) Py_LIMITED_API=0x03030000
-    VBoxPython3m_INCS     = $(VBoxPythonBase_m_INCS) $(VBOX_PYTHON$(VBOX_PYTHON_LIMITED_API_VER)_INC)
-   endif
   endif
  endif # VBOX_WITH_PYTHON_LIMITED_API
 
